### DISCUSSION TITLE
C#

### DISCUSSION THREAD

**Discussion by unicomp23** [connectrpc/connect-es#498] [Ideas]
Am I wrong in thinking we could generate C# as well?  Using this repo?

**Comment by smaye81** [comment]
There is no C# implementation for Connect at the moment, so this repo doesn't support that.  If you mean just generating base types or gRPC artifacts from Protobuf files, there are a few C# plugins out there to do so, but nothing in this repo, no.

**Comment by timostamm** [comment]
ASP.NET has an excellent [middleware to enable gRCP-web](https://www.nuget.org/packages/Grpc.AspNetCore.Web). You can clients from this repo with the [gRCP-web transport](https://connect.build/docs/web/choosing-a-protocol#grpc-web).

**Comment by unicomp23** [comment]
For what it's worth, this little trick helped me stop pulling out my hair while using webstorm.  Would be nice if the codegen could add it automatically somehow.

https://github.com/Thinkei/decentralized-proto-poc/blob/790e161b23182c1aa1a054065bda399a80858186/packages/bar-service/src/handlers/bar.ts#L49

**Reply by timostamm** [reply]
See https://github.com/bufbuild/connect-es/issues/494#issuecomment-1446585003

**Comment by AlanEwaltOnX** [comment]
I'm sorry, but I'm not sure which repo you are referring to. Can you please provide more context or information so I can better understand the question and provide a helpful response? Thank you.
